corehq/apps/userreports/models.py

```diff
@@ -464,6 +470,22 @@ def get_report_count(self):
         """
         return ReportConfiguration.count_by_data_source(self.domain, self._id)
 
+    def validate_db_config(self):
+        mirrored_engine_ids = self.mirrored_engine_ids
+        if not mirrored_engine_ids:
+            return
+        if self.engine_id in mirrored_engine_ids:
+            raise BadSpecError("mirrored_engine_ids list should not contain engine_id")
+
+        for engine_id in mirrored_engine_ids:
+            if not connection_manager.engine_id_is_available(engine_id):
+                raise BadSpecError(
+                    "DB for engine_id {} is not availble".format(engine_id)
+                )
+
+        if not connection_manager.resolves_to_unique_dbs(mirrored_engine_ids + [self.engine_id]):
+            raise BadSpecError("No two engine_ids should point to the same database")
```

Comment:
these last two validations would prevent us from rolling this out to `india` before we rolled it out to `tcl` - if we set `mirrored_engine_ids` in the ICDS data sources then the DB settings would need to be on both envs otherwise this would error.

Reply:
(offlined) Following config should allow us to test on just india first

```
mirrored_engine_ids: [
{'server_env': 'india', 'engine_ids': ['citus']},
{'server_env': 'icds', 'engine_ids': []}
]
```